File: Functions/Lambda_d1rs.cpp
#include<iostream>
#include<cmath>
#include</home/smm5969/Desktop/cpp/Eigen/Eigen/Dense>
//#include <C:\Users\smm5969\Box Sync\Fibrin CPP\Functions\cross_vM.cpp>
using namespace Eigen;
using namespace std;
Matrix3d Lambda_d1rs( Vector3d &T, Vector3d &T1, Vector3d &t, Vector3d &t1, Vector3d &tr, Vector3d &ts, Vector3d &t1r, Vector3d &t1s, Vector3d &trs, Vector3d &t1rs )
{
	double dTt,dT1t,dTt1,dTts,dTtr,dTtrs,dTt1s,dTt1r,dTt1rs,dT1ts,dT1tr,dT1trs; 
	Vector3d CTt,CT1t,CTt1,CTts,CTtr,CTtrs,CTt1s,CTt1r,CTt1rs,CT1ts,CT1tr,CT1trs;
	Vector3d vec_sum1;
	Matrix3d matrix,I,m1,m2,m3,m4,m5,m6,m7,m8,m9,m10,m11,m12; 
	
	dTt=T.dot(t);dT1t=T1.dot(t);dTt1=T.dot(t1);dTts=T.dot(ts);dTtr=T.dot(tr);dTtrs=T.dot(trs);
	dTt1s=T.dot(t1s);dTt1r=T.dot(t1r);dTt1rs=T.dot(t1rs);dT1ts=T1.dot(ts);dT1tr=T1.dot(tr);dT1trs=T1.dot(trs);
	
	
	CTt=T.cross(t);CT1t=T1.cross(t);CTt1=T.cross(t1);CTts=T.cross(ts);CTtr=T.cross(tr);CTtrs=T.cross(trs);
	CTt1s=T.cross(t1s);CTt1r=T.cross(t1r);CTt1rs=T.cross(t1rs);CT1ts=T1.cross(ts);CT1tr=T1.cross(tr);CT1trs=T1.cross(trs);
	
	
	I = Matrix3d::Identity(); 
	
	vec_sum1 = (CT1trs+CTt1rs);
	
	m1=(dT1trs+dTt1rs)*I+cross_vM(vec_sum1,I);
	m2=-6*(((dT1t+dTt1)*dTtr*dTts)/pow(1+dTt,4))*CTt*CTt.transpose();
	m3=((2*(dT1ts+dTt1s)*dTtr+2*(dT1t+dTt1)*dTtrs)/pow(1+dTt,3))*CTt*CTt.transpose();
	m4=(2*(dT1t+dTt1)*dTtr/pow(1+dTt,3))*(CTt*CTts.transpose()+CTts*CTt.transpose());
	m5=(2*(dT1tr+dTt1r)*dTts/pow(1+dTt,3)-(dT1trs+dTt1rs)/pow(1+dTt,2))*CTt*CTt.transpose();
	m6=-((dT1tr+dTt1r)/pow(1+dTt,2))*(CTt*CTts.transpose()+CTts*CTt.transpose());
	m7=(2*(dT1t+dTt1)*dTts/pow(1+dTt,3)-(dT1ts+dTt1s)/pow(1+dTt,2))*(CTt*CTtr.transpose()+CTtr*CTt.transpose());
	m8=-((dT1t+dTt1)/pow(1+dTt,2))*(CTt*CTtrs.transpose()+CTts*CTtr.transpose()+CTtr*CTts.transpose()+CTtrs*CTt.transpose());
	m9=(2*dTtr*dTts/pow(1+dTt,3)-dTtrs/pow(1+dTt,2))*(CTt*(CT1t+CTt1).transpose()+(CT1t+CTt1)*CTt.transpose());
	m10=-(dTtr/pow(1+dTt,2))*(CTt*(CT1ts+CTt1s).transpose()+CTts*(CT1t+CTt1).transpose()+(CT1t+CTt1)*CTts.transpose()+(CT1ts+CTt1s)*CTt.transpose());
	m11=-(dTts/pow(1+dTt,2))*(CTt*(CT1tr+CTt1r).transpose()+CTtr*(CT1t+CTt1).transpose()+(CT1t+CTt1)*CTtr.transpose()+(CT1tr+CTt1r)*CTt.transpose());
	m12=(1/(1+dTt))*(CTt*(CT1trs+CTt1rs).transpose()+CTts*(CT1tr+CTt1r).transpose()+CTtr*(CT1ts+CTt1s).transpose()+CTtrs*(CT1t+CTt1).transpose()+(CT1t+CTt1)*CTtrs.transpose()+(CT1ts+CTt1s)*CTtr.transpose()+(CT1tr+CTt1r)*CTts.transpose()+(CT1trs+CTt1rs)*CTt.transpose());
	
	matrix=m1+m2+m3+m4+m5+m6+m7+m8+m9+m10+m11+m12;

	return matrix;
}

File: Matlab/Formulation/German Formulation/With Torsion/Functions_RN Parallel/3D_New/Lambda_d1rs.m
function [ matrix ] = Lambda_d1rs( T,T1,t,t1,tr,ts,t1r,t1s,trs,t1rs )
%UNTITLED4 Summary of this function goes here
%   Detailed explanation goes here
%% Code
% I=eye(3);
I=[1,0,0;0,1,0;0,0,1];

%% Scalars
% dTt=dot_p(T,t);% Scalar
% dT1t=dot_p(T1,t);dTt1=dot_p(T,t1);
% dTts=dot_p(T,ts);dTtr=dot_p(T,tr);dTtrs=dot_p(T,trs);
% dTt1s=dot_p(T,t1s);dTt1r=dot_p(T,t1r);dTt1rs=dot_p(T,t1rs);
% dT1ts=dot_p(T1,ts);dT1tr=dot_p(T1,tr);dT1trs=dot_p(T1,trs);
dTt=T(1)*t(1)+T(2)*t(2)+T(3)*t(3);dT1t=T1(1)*t(1)+T1(2)*t(2)+T1(3)*t(3);dTt1=T(1)*t1(1)+T(2)*t1(2)+T(3)*t1(3);dTts=T(1)*ts(1)+T(2)*ts(2)+T(3)*ts(3);dTtr=T(1)*tr(1)+T(2)*tr(2)+T(3)*tr(3);dTtrs=T(1)*trs(1)+T(2)*trs(2)+T(3)*trs(3);dTt1s=T(1)*t1s(1)+T(2)*t1s(2)+T(3)*t1s(3);dTt1r=T(1)*t1r(1)+T(2)*t1r(2)+T(3)*t1r(3);dTt1rs=T(1)*t1rs(1)+T(2)*t1rs(2)+T(3)*t1rs(3);dT1ts=T1(1)*ts(1)+T1(2)*ts(2)+T1(3)*ts(3);dT1tr=T1(1)*tr(1)+T1(2)*tr(2)+T1(3)*tr(3);dT1trs=T1(1)*trs(1)+T1(2)*trs(2)+T1(3)*trs(3);

%% Column Vetors
% CTt=cross(T,t);% Column vector.
% CT1t=cross(T1,t);CTt1=cross(T,t1);
% CTts=cross(T,ts);CTtr=cross(T,tr);CTtrs=cross(T,trs);
% CTt1s=cross(T,t1s);CTt1r=cross(T,t1r);CTt1rs=cross(T,t1rs);
% CT1ts=cross(T1,ts);CT1tr=cross(T1,tr);CT1trs=cross(T1,trs);
CTt=cross(T,t);CT1t=cross(T1,t);CTt1=cross(T,t1);CTts=cross(T,ts);CTtr=cross(T,tr);CTtrs=cross(T,trs);CTt1s=cross(T,t1s);CTt1r=cross(T,t1r);CTt1rs=cross(T,t1rs);CT1ts=cross(T1,ts);CT1tr=cross(T1,tr);CT1trs=cross(T1,trs);

%% Matrices
% m1=(dT1trs+dTt1rs)*I+cross_vM((CT1trs+CTt1rs),I);
% m2=-6*(((dT1t+dTt1)*dTtr*dTts)/(1+dTt)^4)*CTt*transpose(CTt);
% m3=((2*(dT1ts+dTt1s)*dTtr+2*(dT1t+dTt1)*dTtrs)/(1+dTt)^3)*CTt*transpose(CTt);
% m4=(2*(dT1t+dTt1)*dTtr/(1+dTt)^3)*(CTt*transpose(CTts)+CTts*transpose(CTt));
% m5=(2*(dT1tr+dTt1r)*dTts/(1+dTt)^3-(dT1trs+dTt1rs)/(1+dTt)^2)*CTt*transpose(CTt);
% m6=-((dT1tr+dTt1r)/(1+dTt)^2)*(CTt*transpose(CTts)+CTts*transpose(CTt));
% m7=(2*(dT1t+dTt1)*dTts/(1+dTt)^3-(dT1ts+dTt1s)/(1+dTt)^2)*(CTt*transpose(CTtr)+CTtr*transpose(CTt));
% m8=-((dT1t+dTt1)/(1+dTt)^2)*(CTt*transpose(CTtrs)+CTts*transpose(CTtr)+CTtr*transpose(CTts)+CTtrs*transpose(CTt));
% m9=(2*dTtr*dTts/(1+dTt)^3-dTtrs/(1+dTt)^2)*(CTt*transpose(CT1t+CTt1)+(CT1t+CTt1)*transpose(CTt));
% m10=-(dTtr/(1+dTt)^2)*(CTt*transpose(CT1ts+CTt1s)+CTts*transpose(CT1t+CTt1)+(CT1t+CTt1)*transpose(CTts)+(CT1ts+CTt1s)*transpose(CTt));
% m11=-(dTts/(1+dTt)^2)*(CTt*transpose(CT1tr+CTt1r)+CTtr*transpose(CT1t+CTt1)+(CT1t+CTt1)*transpose(CTtr)+(CT1tr+CTt1r)*transpose(CTt));
% m12=(1/(1+dTt))*(CTt*transpose(CT1trs+CTt1rs)+CTts*transpose(CT1tr+CTt1r)+CTtr*transpose(CT1ts+CTt1s)+CTtrs*transpose(CT1t+CTt1)+(CT1t+CTt1)*transpose(CTtrs)+(CT1ts+CTt1s)*transpose(CTtr)+(CT1tr+CTt1r)*transpose(CTts)+(CT1trs+CTt1rs)*transpose(CTt));
% matrix=m1+m2+m3+m4+m5+m6+m7+m8+m9+m10+m11+m12;

matrix=(dT1trs+dTt1rs)*I+cross_vM((CT1trs+CTt1rs),I)-6*(((dT1t+dTt1)*dTtr*dTts)/(1+dTt)^4)*CTt*transpose(CTt)+((2*(dT1ts+dTt1s)*dTtr+2*(dT1t+dTt1)*dTtrs)/(1+dTt)^3)*CTt*transpose(CTt)+(2*(dT1t+dTt1)*dTtr/(1+dTt)^3)*(CTt*transpose(CTts)+CTts*transpose(CTt))+(2*(dT1tr+dTt1r)*dTts/(1+dTt)^3-(dT1trs+dTt1rs)/(1+dTt)^2)*CTt*transpose(CTt)-((dT1tr+dTt1r)/(1+dTt)^2)*(CTt*transpose(CTts)+CTts*transpose(CTt))+(2*(dT1t+dTt1)*dTts/(1+dTt)^3-(dT1ts+dTt1s)/(1+dTt)^2)*(CTt*transpose(CTtr)+CTtr*transpose(CTt))-((dT1t+dTt1)/(1+dTt)^2)*(CTt*transpose(CTtrs)+CTts*transpose(CTtr)+CTtr*transpose(CTts)+CTtrs*transpose(CTt))+(2*dTtr*dTts/(1+dTt)^3-dTtrs/(1+dTt)^2)*(CTt*transpose(CT1t+CTt1)+(CT1t+CTt1)*transpose(CTt))-(dTtr/(1+dTt)^2)*(CTt*transpose(CT1ts+CTt1s)+CTts*transpose(CT1t+CTt1)+(CT1t+CTt1)*transpose(CTts)+(CT1ts+CTt1s)*transpose(CTt))-(dTts/(1+dTt)^2)*(CTt*transpose(CT1tr+CTt1r)+CTtr*transpose(CT1t+CTt1)+(CT1t+CTt1)*transpose(CTtr)+(CT1tr+CTt1r)*transpose(CTt))+(1/(1+dTt))*(CTt*transpose(CT1trs+CTt1rs)+CTts*transpose(CT1tr+CTt1r)+CTtr*transpose(CT1ts+CTt1s)+CTtrs*transpose(CT1t+CTt1)+(CT1t+CTt1)*transpose(CTtrs)+(CT1ts+CTt1s)*transpose(CTtr)+(CT1tr+CTt1r)*transpose(CTts)+(CT1trs+CTt1rs)*transpose(CTt));
end
